### PR TITLE
Renaming Query->Request and Result->Response

### PR DESCRIPTION
**Pull request by khieta**
*Issue #, if available:*

*Description of changes:*

Renamed `ValidationQuery` -> `ValidationRequest` and `ValidationResult` -> `ValidationResponse` to match `AuthorizationRequest` and `AuthorizationResponse`.

And since I was already messing with things, I made a few other non-functional changes too:
* Renamed variables for consistency (e.g., `query` -> `request`)
* Updated terminology in comments
* Fixed a couple typos
* Removed some unused imports

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by john-h-kastner-aws — APPROVED** [review]
Looks good

File: CedarJava/src/main/java/com/cedarpolicy/AuthorizationEngine.java [hunk 1]
```diff
@@ -37,30 +37,30 @@
...
+     * @param request The request to evaluate
      * @param slice The slice to evaluate against
-     * @return The result of the query evaluation
-     * @throws AuthException On failure to make the authorization query. Note that errors inside the
+     * @return The result of the request evaluation
+     * @throws AuthException On failure to make the authorization request. Note that errors inside the
      *     authorization engine are included in the <code>errors</code> field on the
-     *     AuthorizationResult. Note: This error interface will likely change in the future. We will
+     *     AuthorizationResponse. Note: This error interface will likely change in the future. We will
      *     likely unify the error handling story.
```
Comment (cdisselkoen) [inline CedarJava/src/main/java/com/cedarpolicy/AuthorizationEngine.java:49]: is this still true

File: CedarJava/src/main/java/com/cedarpolicy/AuthorizationEngine.java [hunk 2]
```diff
@@ -37,30 +37,30 @@
...
      */
-    AuthorizationResponse isAuthorized(AuthorizationRequest q, Slice slice) throws AuthException;
+    AuthorizationResponse isAuthorized(AuthorizationRequest request, Slice slice) throws AuthException;
```
Comment (cdisselkoen) [inline CedarJava/src/main/java/com/cedarpolicy/AuthorizationEngine.java:51]: Should we rename `Slice` as well, to match the Rust API better

**Review by cdisselkoen — APPROVED** [review]

File: CedarJava/src/main/java/com/cedarpolicy/AuthorizationEngine.java (hunk 2 above)
Comment (khieta) [reply CedarJava/src/main/java/com/cedarpolicy/AuthorizationEngine.java:50]: Opened a new issue for this #40 (if I misinterpreted your suggestion, feel free to comment on the issue or edit it directly)

Comment (cdisselkoen) [reply CedarJava/src/main/java/com/cedarpolicy/AuthorizationEngine.java:50]: Nope, that's exactly what I meant, looks great

**Pull request merged**